sql/core/src/main/scala/org/apache/spark/sql/SQLImplicits.scala

```diff
@@ -74,6 +74,9 @@ abstract class SQLImplicits {
   /** @since 1.6.0 */
   implicit def newStringEncoder: Encoder[String] = Encoders.STRING
 
+  /** @since 2.2.0 */
+  implicit def newDecimalEncoder: Encoder[java.math.BigDecimal] = Encoders.DECIMAL
+
```

Comment:
Thanks for the feedback, @cloud-fan @HyukjinKwon. I will update the PR to add more encoders.